### ISSUE TITLE
Column width setting is not working!?!

### ISSUE THREAD

**Issue by sejokine** [gitextensions/gitextensions#3603]
--- GitExtensions2.49.01, Git 2.10.1.windows.1, Microsoft Windows NT 6.3.9600.0

My OS is Windows 10.

**Comment by pmiossec** [comment]
Thanks for this wonderful explanation of what happens and what is wrong that will help fix the problem! 

PS: and I also love your use of '!?!' that prove that it's really urgent and will push us to fix it quickly..... (I prefer '...' )

**Comment by manu-vilachan** [comment]
@pmiossec I've got the same problem, but I'll try to explain better.

In the main view with a repo opened, the last column it's impossible to resize. In my case, I try to show sha1 column and it's impossible to make it larger, but yes to make it smaller.
![gitextensions-screenshot](https://cloud.githubusercontent.com/assets/9888749/24307190/6888dcb6-10c3-11e7-9ef0-be0893d07256.png)

What happens to me, it's that trying to resize it the las column it gets too smaller and I cannot expand it again, despite the resize cursor appears at the end of the column (I cannot capture cursor image but appears).

![gitextensions-screenshot](https://cloud.githubusercontent.com/assets/9888749/24307190/6888dcb6-10c3-11e7-9ef0-be0893d07256.png)

Ahh, I try it in W10 & W8.1, with versions 2.49.01 & 2.49.02, same behavior.

Hope it helps!

Regards.

**Comment by vbjay** [comment]
I'm wondering if a reset current layout widths feature would be worth
implementing?  Fyi. There are more than one layouts.

**Comment by EbenZhang** [comment]
not sure will #3573 help or not, it is in master, hasnt been released yet

**Comment by glen-nicol** [comment]
I have also noticed this on the subject column in the rebase dialog.

**Comment by RussKie** [comment]
dup of #3447 #3580

**Comment by RussKie** [comment]
Can you please verify if it is still an issue in the latest version?

**Comment by iinnkkyy** [comment]
Yes, it is still a problem. But the workaround works.

**Comment by RussKie** [comment]
Duplicate of #3384 #3447 #3580

**Comment by RussKie** [comment]
Duplicate of #3384
Duplicate of #3447
Duplicate of #3580

**Comment by tomlogic** [comment]
Still seeing this with 2.51 (note that it is on an hdpi display).  I cannot increase the width of the date field so I can view the time, and it's odd that the ellipsis on the commit message is cropped.

![image](https://user-images.githubusercontent.com/1102448/37870865-2e282760-2f95-11e8-8893-a19e9c0a1e75.png)

**Comment by drewnoakes** [comment]
I believe this was fixed in #4772 with [these changes](https://github.com/gitextensions/gitextensions/pull/4772/files#diff-5bef6f988b99f05b5ae260459a48ffa5).

Can you try the code from `master` or a build from [here](https://ci.appveyor.com/project/gitextensions/gitextensions/branch/master/artifacts) and verify it's fixed for you?

**Comment by tomlogic** [comment]
@drewnoakes Looking forward to the extensive changes from #4772!  I went to the appveyor link and didn't see artifacts on the recent builds, and I don't have a build system set up for building from `master`.  I'll keep an eye out for the next 2.51 release and report back on whether it's working or not.

**Comment by RussKie** [comment]
@tomlogic here's the build for #4772 - https://ci.appveyor.com/api/buildjobs/lh5ywvy7443dvea0/artifacts/GitExtensions-Portable-3.00.00.2346.zip
Please try it and see if the specific issue is fixed.

The usual disclaimer - this is a canary build, and whilst it may be usable it may not be entirely stable.

**Comment by tomlogic** [comment]
@RussKie Looking good so far.  Columns sized correctly.  I don't usually have SHA1 column visible, and I did run into a problem of not being able to widen the date column after a test to narrow it (can't extend the right edge of the right-most column).  I had to show the SHA1 column before I could make the date column wider.

The clone dialog clips the two buttons on the bottom of the screen, but I see that covered by #4776.  I'll try using that build and grab screenshots of anything that isn't covered by #4775.
